### PR TITLE
provide configuration parameters to a monitoring connection

### PR DESCRIPTION
**Pull request by sergiyv-bitquill**
### Summary

Check if Monitor Requires Additional Properties

### Description

Provide a way to supply monitoring connection with a custom set of configuration properties.

### Additional Reviewers

@karenc-bq 
@ColinKYuen 
@brunos-bq 
@matthewh-BQ 
@yanw-bq

### CODE REVIEW

File: src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/plugins/Monitor.java [hunk 1]
```diff
@@ -154,11 +155,10 @@ ConnectionStatus checkConnectionStatus(final int shortestFailureDetectionInterva
       if (this.monitoringConn == null || this.monitoringConn.isClosed()) {
         // open a new connection
         Map<String, String> properties = new HashMap<>();
```
Comment (brunos-bq) [inline src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/plugins/Monitor.java:157]: A bit confusing with properties and props.

I suggest calling this map "propertyMap"

File: src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/plugins/Monitor.java [hunk 2]
```diff
@@ -55,6 +55,7 @@
   }
 
   private static final int THREAD_SLEEP_WHEN_INACTIVE_MILLIS = 100;
+  private static final String MONITORING_CONNECTION_PROPERTY_PREFIX = "monitoring-";
```
Comment (brunos-bq) [inline src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/plugins/Monitor.java:58]: NIT: Can drop CONNECTION from the name maybe?

File: src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/plugins/Monitor.java [hunk 3]
```diff
@@ -154,11 +155,10 @@ ConnectionStatus checkConnectionStatus(final int shortestFailureDetectionInterva
       if (this.monitoringConn == null || this.monitoringConn.isClosed()) {
         // open a new connection
         Map<String, String> properties = new HashMap<>();
-        properties.put(PropertyKey.tcpKeepAlive.getKeyName(),
-            this.propertySet.getBooleanProperty(PropertyKey.tcpKeepAlive).getStringValue());
-        properties.put(PropertyKey.connectTimeout.getKeyName(),
-            this.propertySet.getBooleanProperty(PropertyKey.connectTimeout).getStringValue());
-        //TODO: any other properties to pass? like socket factory
+        Properties props = this.propertySet.exposeAsProperties();
```
Comment (brunos-bq) [inline src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/plugins/Monitor.java:158]: Than this one would be "properties"

File: src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/plugins/Monitor.java [hunk 4]
```diff
@@ -154,11 +155,10 @@ ConnectionStatus checkConnectionStatus(final int shortestFailureDetectionInterva
...
+        props.stringPropertyNames().stream()
+          .filter(propName -> propName.startsWith(MONITORING_CONNECTION_PROPERTY_PREFIX))
```
Comment (brunos-bq) [inline src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/plugins/Monitor.java:160]: NIT: i'd replace propName with "p", just cause it's in the lambda expression

File: src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/plugins/Monitor.java (hunk 1 above)
Comment (sergiyv-bitquill) [reply src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/plugins/Monitor.java]: Renamed it to monitoringConnProperties. I hope it's clearer.

File: src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/plugins/Monitor.java (hunk 3 above)
Comment (sergiyv-bitquill) [reply src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/plugins/Monitor.java]: Renamed it to originalProperties.

**Review by brunos-bq — APPROVED** [review]
Much better.

Very easy now to spot right away what are the properties that we have fetched and the ones that we are passing forward.

**Pull request merged**